packages/hyperion-util/src/onNetworkRequest.ts

```diff
@@ -3,58 +3,80 @@
  */
 
 import { Hook } from "@hyperion/hook";
-import * as IWindow from "@hyperion/hyperion-dom/src/IWindow";
+// import * as IWindow from "@hyperion/hyperion-dom/src/IWindow";
 import * as IXMLHttpRequest from "@hyperion/hyperion-dom/src/IXMLHttpRequest";
 import * as intercept from "@hyperion/hyperion-core/src/intercept";
 import { assert } from "@hyperion/global";
 
-type NetworkRequest = {
+type NetworkRequestInfo = {
   readonly body?: RequestInit['body'],
   readonly method: string;
-  readonly url: string | URL; // TODO: should we always send on of the types?
+  readonly url: string;
 }
 
 /**
  * This is a generic event to be fired when a network request is about to happen.
  * Generally, various network api can be intercepted to provide a more unified way
  * of notifying the app about network requests.
  */
-export const onNetworkRequest = new Hook<(request: NetworkRequest) => void>();
-
-IWindow.fetch.onArgsObserverAdd((input, init) => {
-  let request: NetworkRequest;
-  if (typeof input === "string") {
-    request = {
-      body: init?.body,
-      method: init?.method ?? "get",
-      url: input,
-    };
-  } else if (input instanceof Request) {
-    request = {
-      body: input.body,
-      method: input.method,
-      url: input.url,
-    }
-  } else {
-    request = {
-      method: "GET",
-      url: input
-    }
-  }
+export const onNetworkRequestSend = new Hook<(request: XMLHttpRequest, requestInfo: NetworkRequestInfo) => void>();
```

Comment:
If we are making everything very XHR specific, then do we even need to have yet another set of events for "onNetwork` defined here? Wouldn't it be easier to just expose XHR?

Also, alternatively, I think we can just add an `initiatorXHR?: XMLHttpRequest` to the `NetworkRequestInfo` itself and continue keeping it generic.